### PR TITLE
Github actions updates for #13

### LINKED ISSUE #13: Limit github actions publishing to pushes to main.

### PR DESCRIPTION
**Pull request by pmanko**
Closes #13

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -10,16 +10,12 @@ on:
...
   push:
     branches: [ main ]
-  pull_request:
-    branches: [ main ]
 
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
-jobs:
```
Comment (mozzy11) [inline .github/workflows/publish.yml:20]: hi @pmanko ,did you just  remove the `jobs:`  accidentally ??
wont the ci build fail ??

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        uses: docker://hl7fhir/ig-publisher-base:latest
+        with:
+          # Get the latest publisher - don't run the batch script but run the line directly
+          args: curl -L https://github.com/HL7/fhir-ig-publisher/releases/latest/download/publisher.jar -o ./input-cache/publisher.jar --create-dirs
+
+      - name: Run the IG publisher
+        uses: docker://hl7fhir/ig-publisher-base:latest
+        with:
+          # Run the publisher - don't run the batch script but run the line directly
+          args: java -jar ./input-cache/publisher.jar publisher -ig .
```
Comment (mozzy11) [inline .github/workflows/test.yml:38]: so this simply just builds the IG without deploying it anywhere, how would we visualize the changes pushed on given branch ??

Comment (mozzy11) [reply .github/workflows/test.yml:38]: I think as i comented in slack , 
we can still build the changes on a given branch and still deploy them to gh-pages  , in order to vusalise the changes in the IG ,
but then re-run the Actions on the main branch to Update the IG to the true version

Comment (pmanko) [reply .github/workflows/test.yml:38]: I think it would make sense to keep the IG build on PRs, since it is a CI version of the IG.

Comment (pmanko) [reply .github/workflows/test.yml:38]: However, we should probably also have a more official location for hosting a release of the IG

Comment (pmanko) [reply .github/workflows/test.yml:38]: So then, only releases would trigger a publication to that location, but we could keep the original full build on PRs

Comment (mozzy11) [reply .github/workflows/test.yml:38]: Yes that makes sense

File: .github/workflows/publish.yml (hunk 1 above)
Comment (mozzy11) [reply .github/workflows/publish.yml:20]: hi @pmanko ,
i guess you removed the **jobs:**  accidentally  . you can just fix this and we merge this PR

**Comment by mozzy11** [comment]
@pmanko i commited the minor fix  directly here to fix this PR , i can now merge it
you had removed the 'jobs:' key word form the publish.yml accidentally

**Pull request merged**